### ISSUE TITLE
Cordova platform add android, removes node modules with semver

### ISSUE THREAD

**Issue by alex-jesper** [apache/cordova-lib#801]
# Bug Report
When running
```
cordova platform add android --save
```
The node modules with a git reference and using semver are removed from the package-lock.json file and from node_modules folder.

I have run npm i, first and the modules are fetched fine. But the cordova command removes the modules and subsequent use of the modules fails.

The git reference and semver looks like this:
```
"dependencies": {
    "project": "https://bitbucket.org/project.git#semver:~v0.1.1"
}
```
## Problem
Could be related to:
[A similar issue I reported](https://github.com/apache/cordova-lib/issues/795)
[An issue @dpogue describes](https://npm.community/t/npm-install-via-cordova-removing-everything-in-node-modules/3354)

### What is expected to happen?
Package-lock.json should be left untouched and the modules not deleted.

### What does actually happen?
Package-lock.json is edited and the modules deleted.

## Information

I removed semver from the git reference and simply use the hash then everything works again.
```
"dependencies": {
    "project": "https://bitbucket.org/project.git#v0.1.1"
}
```

### Command or Code

NA

### Environment, Platform, Device

NA

### Version information

Cordova 9.0.0

## Checklist

- [x] I searched for existing GitHub issues
- [x] I updated all Cordova tooling to most recent version
- [x] I included all the necessary information above

**Comment by raphinesse** [comment]
I could not reproduce this with a Cordova version that includes #827 which fixes #795. 

Could you please try if you can reproduce the problem when using #827? You can get such a version either by using `npm link` or by waiting until #827 is merged and we have a nightly build that includes it (can then be installed by `npm i cordova@nightly`).

**Comment by alex-jesper** [comment]
As indicated in the ticket the bugs are very similar. I am confident that your fix in #795 has fixed this as a side effect.
Once a nightly build with the fix is ready I will try to reproduce.

**Comment by alex-jesper** [comment]
I am not able to reproduce at all. It must have been some special conditions I experienced the issue.
